Framework/Tools/Emulator/Ssl.cs

```diff
@@ -608,9 +650,13 @@ public int DataAvailable( int socket )
 
             if (!GetSslData(socket, out ssd)) return (int)SocketError.SocketError;
 
-            if(!_socketsDriver.GetSocket(socket, out sock)) return (int)SocketError.SocketError;
+            if (!_socketsDriver.GetSocket(socket, out sock)) return (int)SocketError.SocketError;
 
-            return sock.Poll(0,SelectMode.SelectRead) ? 1024 : 0;
+            var readData = ssd.m_readData;
+            int ret = readData != null ?

```

Comment:
Please split this up into If/else it's pretty dense and anything that forces the reader to have to stop and think, just to parse the code let alone understand its intent is something to avoid.